service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ObservationStatementMapper.java

```diff
@@ -1,35 +1,116 @@
 package uk.nhs.adaptors.gp2gp.ehr.mapper;
 
-import static uk.nhs.adaptors.gp2gp.ehr.utils.DateFormatUtil.toHl7Format;
+import java.util.List;
+import java.util.Optional;
 
+import org.apache.commons.lang3.StringUtils;
+import org.hl7.fhir.dstu3.model.Attachment;
 import org.hl7.fhir.dstu3.model.Observation;
+import org.hl7.fhir.dstu3.model.Quantity;
 import org.hl7.fhir.dstu3.model.ResourceType;
+import org.hl7.fhir.dstu3.model.SampledData;
+import org.hl7.fhir.dstu3.model.Type;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 import com.github.mustachejava.Mustache;
+import com.google.common.collect.ImmutableList;
 
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import uk.nhs.adaptors.gp2gp.ehr.mapper.parameters.ObservationStatementTemplateParameters;
+import uk.nhs.adaptors.gp2gp.ehr.utils.DateFormatUtil;
 import uk.nhs.adaptors.gp2gp.ehr.utils.StatementTimeMappingUtils;
 import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
 
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 @Component
+@Slf4j
 public class ObservationStatementMapper {
+    private static final List<Class<? extends Type>> UNHANDLED_TYPES = ImmutableList.of(SampledData.class, Attachment.class);
     private static final Mustache OBSERVATION_STATEMENT_EFFECTIVE_TIME_TEMPLATE =
-        TemplateUtils.loadTemplate("ehr_observation_statement_effective_time_template.mustache");
+        TemplateUtils.loadTemplate("unstructured_observation_statement_template.mustache");
+    private static final String REFERENCE_RANGE_UNIT_PREFIX = "Range Units: ";
+
     private final MessageContext messageContext;
+    private final StructuredObservationValueMapper structuredObservationValueMapper;
+    private final PertinentInformationObservationValueMapper pertinentInformationObservationValueMapper;
 
     public String mapObservationToObservationStatement(Observation observation, boolean isNested) {
-        var observationStatementTemplateParameters = ObservationStatementTemplateParameters.builder()
+        String comment = observation.getComment();
```

Comment:
What do you think about reversing the mapper logic a bit. It would be more intuitive for me, if there code related to comment, range units and more.. were in separated code chuncks; with separated methods buildComment , buildStructuredElement and all other.

Reply:
The problem is that we either construct a comment according to value type in the observation resource or the value element. This is a little bit complicated because we also prepend comment data if we have a reference range present. I am not 100% sure if this solution would make the code more readable or more complicated.